src/compiler/checker.ts

```diff
@@ -10702,8 +10779,24 @@ namespace ts {
                 }
 
                 let derived = getTargetSymbol(getPropertyOfObjectType(type, base.name));
-                if (derived) {
-                    let baseDeclarationFlags = getDeclarationFlagsFromSymbol(base);
+                let baseDeclarationFlags = getDeclarationFlagsFromSymbol(base);
+
+                Debug.assert(!!derived, "derived should point to something, even if it is the base class' declaration.");
+
+                if (derived === base) { 

```

Comment:
So the case we discussed was where the base class has some generic type parameters and the abstract method uses some of those generic type parameters. The question was whether `derived === base` holds in the case where the derived class actually instantiates the type parameters but doesn't implement/redeclare the method.

Based on the following test, it appears the it does hold:

```ts
abstract class A<T> {
    abstract foo(): T;
}

class B extends A<number> {} // reports an error about not implementing `foo()`
```